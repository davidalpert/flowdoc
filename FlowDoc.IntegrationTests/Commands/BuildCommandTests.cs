using System;
using System.Diagnostics;
using System.IO;
using System.Linq;
using FlowDoc.IntegrationTests.Helpers;
using NUnit.Framework;
using ObjectApproval;

namespace FlowDoc.IntegrationTests.Commands
{
    [TestFixture]
    public class BuildCommandTests
    {
        [Test]
        public void Execute_against_an_empty_directory()
        {
            var currentDir = Environment.CurrentDirectory;
            var flowExe = Path.Combine(currentDir, "flow.exe");
            Assert.IsTrue(File.Exists(flowExe));

            using (var targetPath = new DisposableTempFolder())
            using (var console = new RedirectedConsole())
            {
                var args = new string[] {"build", targetPath};

                var result = CLI.Program.Main(args);

                Debug.WriteLine(console.Output);

                Assert.AreEqual(-1, result);
            }
        }

        [Test]
        public void Execute_against_a_source_directory()
        {
            var currentDir = Environment.CurrentDirectory;
            var flowExe = Path.Combine(currentDir, "flow.exe");
            Assert.IsTrue(File.Exists(flowExe));

            using (var targetPath = new DisposableTempFolder())
            using (var console = new RedirectedConsole())
            {
                CLI.Program.Main("init", targetPath);

                var result = CLI.Program.Main("build", targetPath);

                Debug.WriteLine(console.Output);

                Assert.AreEqual(0, result);

                TestHelpers.VerifyFileNames(targetPath);
            }
        }
    }
}