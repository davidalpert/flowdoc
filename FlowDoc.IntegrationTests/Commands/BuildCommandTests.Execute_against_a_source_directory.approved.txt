[
    "~\\.config",
    "~\\toc.md",
    "~\\_build\\index.html"
]